src/main_loop.rs

```diff
@@ -113,10 +115,21 @@ impl Future for MainLoopState {
                     Session::connect(session_config, creds, cache, handle);
             }
 
-            if let Some(ref mut player_event_channel) = self.spotifyd_state.player_event_channel {
-                if let Async::Ready(Some(event)) = player_event_channel.poll().unwrap() {
-                    if let Some(ref program) = self.spotifyd_state.player_event_program {
-                        run_program_on_events(event, program);
```

Comment:
Maybe there's a nice way of expressing the next block, all I want is to set the Option to None if try_wait is successful. With NLL I could rewrite this to be a bit more straight forward, but on stable I haven't seen anything better than taking the value out of the Option and then putting it back.